### ISSUE TITLE
Warning when using fit_generator

### ISSUE THREAD

**Issue by ThinerDAS** [tensorflow/tensorflow#34099]
Thank you for submitting a TensorFlow documentation issue. Per our GitHub
policy, we only address code/doc bugs, performance issues, feature requests, and
build/installation issues on GitHub.

The TensorFlow docs are open source! To get involved, read the documentation
contributor guide: https://www.tensorflow.org/community/contribute/docs

## URL(s) with the issue:

https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit_generator
https://www.tensorflow.org/api_docs/python/tf/keras/Model#predict_generator

### Clear description

I have had pain debugging here so it would be better if the documentation mentions the fallacy here.

My experience is: I used ctypes C++ to generate training data into a fixed numpy buffer and the generator merely invoked C++ method and return the same buffer in each iteration (with different content obviously). This is problematic as the internal implementation of `fit_generator` will iterate in advance. The correct usage is to make a copy of the buffer each time iterator is iterated

The document should probably mention explicitly that the method does not make deep copy of the return values of the generator on its generation, so e.g. **it is probably not a good idea to share a same numpy buffer across different batches - if one want to yield from same buffer due to certain reasons, one good practice is to make a `nparray.copy()` before `yield`**.

**Comment by ravikyram** [comment]
In order to expedite the trouble-shooting process, please provide a minimal standalone code to reproduce the issue reported here. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 44 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
We are closing this issue for now due to lack of activity. Please comment if this is still an issue for you. Thanks!
